PWGHF/D2H/Tasks/taskDirectedFlowCharmHadrons.cxx

```diff
@@ -0,0 +1,304 @@
+      registry.fill(HIST("hpuyQytvscentpteta"), mass, cent, pt, eta, uyQyt, sign);
+    }
+  }
+
+  /// Get the centrality
+  /// \param collision is the collision with the centrality information
+  double getCentrality(CollsWithQvecs::iterator const& collision)
+  {
+    double cent = -999.;
+    switch (centEstimator) {
+      case CentralityEstimator::FV0A:
+        cent = collision.centFV0A();
+        break;
+      case CentralityEstimator::FT0M:
+        cent = collision.centFT0M();
+        break;
+      case CentralityEstimator::FT0A:
+        cent = collision.centFT0A();
+        break;
+      case CentralityEstimator::FT0C:
+        cent = collision.centFT0C();
+        break;
+      default:
+        LOG(warning) << "Centrality estimator not valid. Possible values are V0A, T0M, T0A, T0C. Fallback to V0A";
+        cent = collision.centFV0A();
+        break;
+    }
+    return cent;
+  }
+
+  /// Check if the collision is selected
+  /// \param collision is the collision with the Q vector information
+  /// \param bc is the bunch crossing with timestamp information
+  /// \return true if the collision is selected, false otherwise
+  template <o2::hf_centrality::CentralityEstimator centEstimator>
+  bool isCollSelected(CollsWithQvecs::iterator const& collision,
+                      aod::BCsWithTimestamps const&)
+  {
+    double centrality{-1.f};
+    const auto rejectionMask = hfEvSel.getHfCollisionRejectionMask<true, centEstimator, aod::BCsWithTimestamps>(collision, centrality, ccdb, registry);
+
+    /// monitor the satisfied event selections
+    hfEvSel.fillHistograms(collision, rejectionMask, centrality);
+    return rejectionMask == 0;
+  }
+
+  /// Get the Q vector
+  /// \param collision is the collision with the Q vector information
+  std::vector<double> getQvec(CollsWithQvecs::iterator const& collision)
+  {
+    double qxAQVec = 0.0;
+    double qyAQVec = 0.0;
+    double qxCQVec = 0.0;
+    double qyCQVec = 0.0;
+    qxAQVec = collision.qxZDCA();
+    qyAQVec = collision.qyZDCA();
+    qxCQVec = collision.qxZDCC();
+    qyCQVec = collision.qyZDCC();
+    return {qxAQVec, qyAQVec, qxCQVec, qyCQVec};
+  }
+
+  /// Compute the scalar product
+  /// \param collision is the collision with the Q vector information and event plane
+  /// \param candidates are the selected candidates
+  template <DecayChannel channel, typename T1, typename Trk>
+  void runFlowAnalysis(CollsWithQvecs::iterator const& collision,
+                       T1 const& candidates, Trk const& /*tracks*/)
+  {
+    double cent = getCentrality(collision);
+    if (cent < centralityMin || cent > centralityMax) {
+      return;
+    }
+
+    if (!collision.triggerevent()) { // for selecting only callibrated events
+      return;
+    }
+
+    std::vector<double> qVecs = getQvec(collision);
+    double qxAQVec = qVecs[0];
+    double qyAQVec = qVecs[1];
+    double qxCQVec = qVecs[2];
+    double qyCQVec = qVecs[3];
```

Comment:
Can be assigned directly from `collision`.

Reply:
[Reply]: Done as suggested.